\documentclass{vgtc}                      % final (conference style)
%\documentclass[review]{vgtc}                 % review
%\documentclass[widereview]{vgtc}             % wide-spaced review
%\documentclass[preprint]{vgtc}               % preprint
%\documentclass[electronic]{vgtc}             % electronic version

\ifpdf
  \pdfoutput=1\relax
  \pdfcompresslevel=9 
  \pdfoptionpdfminorversion=7
  \ExecuteOptions{pdftex}
  \usepackage{graphicx}
  \DeclareGraphicsExtensions{.pdf,.png,.jpg,.jpeg} 
\else%
  \ExecuteOptions{dvips}
  \usepackage{graphicx}
  \DeclareGraphicsExtensions{.eps}
\fi

%% it is recomended to use ``\autoref{sec:bla}'' instead of ``Fig.~\ref{sec:bla}''
\graphicspath{{figures/}{pictures/}{images/}{./}}

\usepackage{microtype}
\PassOptionsToPackage{warn}{textcomp} 
\usepackage{textcomp}
\usepackage{amsmath}
\usepackage{mathptmx}
\usepackage{times}
\renewcommand*\ttdefault{txtt}
\usepackage{cite}
\usepackage{tabu} 
\usepackage{booktabs}

%% If you are submitting a paper to a conference for review with a double
%% blind reviewing process, please replace the value ``0'' below with your
%% OnlineID. Otherwise, you may safely leave it at ``0''.
\onlineid{0}

%% declare the category of your paper, only shown in review mode
\vgtccategory{Research}

%% allow for this line if you want the electronic option to work properly
\vgtcinsertpkg

%% In preprint mode you may define your own headline.
%\preprinttext{To appear in an IEEE VGTC sponsored conference.}

%% Paper title.
\title{SynMapN: Interactive Visual Comparison for Multiple Genomes}

\author{
Mingwei Li\thanks{e-mail: mwli@email.arizona.edu}
\and Asher Keith Haug-Baltzell\thanks{email: ahaug@email.arizona.edu}
\and Eric Lyons\thanks{email: elyons.uoa@gmail.com}
\and Carlos Scheidegger\thanks{e-mail: cscheid@cs.arizona.edu}
%TODO Eric and Asher, Heather, Sean?
}
\affiliation{\scriptsize University of Arizona}

%% A teaser figure can be included as follows, but is not recommended since
%% the space is now taken up by a full width abstract.
%\teaser{
%  \includegraphics[width=1.5in]{sample.eps}
% \caption{Lookit! Lookit!}
%}

%% Abstract section.
\abstract{
  % First, set up the context. (``why'')
  % Then, say what.
  % Then, how,
  Interactive visualization has become a powerful means to explore syntenic relationships among two genomes, with a variety of available tools for domain scientists to employ.
  However, these tools do not tend to scale well in the case where \emph{many} genomes are compared against one another.
  This poster describes ongoing efforts to build techniques and tools to help genomicists understand sets of genomes and their syntenic relationships.
  %% We try to answer the question of how to provide visual guidance in studying the genetic relations of many genomes.
  %% In this poster we give a study of visual exploration of data in the area of comparative genomics.
  Our main contribution is a mechanism that defines \emph{set distances}: this can be used to compare entire genomes to one another, as well as \emph{sets of genomes} to each other.
  Currently, we use this mechanism to generate dimensionality reduction visualizations. We discuss limitations of this approach, as well as future directions.
}

%% ACM Computing Classification System (CCS). 
%% See <http://www.acm.org/class/1998/> for details.
%% The ``\CCScat'' command takes four arguments.
%TODO how to generate this
\CCScatlist{ 
  %\CCScat{K.6.1}{Management of Computing and Information Systems}%
%{Project and People Management}{Life Cycle};
  %\CCScat{K.7.m}{The Computing Profession}{Miscellaneous}{Ethics}
}


%% Copyright space is enabled by default as required by guidelines.
%% It is disabled by the 'review' option or via the following command:
% \nocopyrightspace


%%%%%%%% START OF THE PAPER %%%%%%%%%%%%

\begin{document}
\firstsection{Introduction}
\maketitle
%% \section{Introduction} %for journal use above \firstsection{..} instead
In comparative genomics, one of the visual methods that helps studying the structural relations between two genomes is called the syntenic dotplot \cite{syntenic_dotplot, synmap}.
It is a scatterplot that depicts matched genes between two genomes; these matched genes imply \emph{syntenic regions}: those that likely originated from the same ancestor.
During genomic analysis, a measure called synonymous mutation rate between two genes ($ks$) is computed.
Because many codons translate to the same amino acid (for example, TCT and TCC both translate to serine), it is possible that a mutation does not change the amino acid that is encoded in the gene: this is a ``synonymous'' mutation.
The $ks$ value is a rate of such mutations, normalized by gene sizes. Because those mutations are mostly harmless, they tend to not change selection pressure on the genomes. At the same time, they are transmitted hereditarily, and so they can be seen as ``biological clocks'', and thus can be used to estimate evolutionary relations.
The larger the $ks$ value is, the longer it has been since these two genomes diverged.
%explain more on the plot, PARAPHRASE this:
%Each axis represents a sequence laid end-to-end, and each dot in the scatter-plot represents a putative homologous match between the two sequences
In the dotplot, two axes represent gene locations of two genomes respectively. Each dot on the plot represents a match between two genes, and the dot colors usually encode $ks$ values.

In some ideal cases, a perfect alignment of two genomes can be observed by seeing a line along the diagonal of a plot, similar to a plot of function $y=x$. This means that the two entities in concern have same genes presented in order. More interesting events like duplication and inversion of genes can be easily spotted from the syntenic dotplot as well (\autoref{fig:dotplot_arabidopsis_marked}). In other cases, it is also common to see only sparse dots, indicating no significant alignment in the two genomes.

Although tools such as syntenic dotplots and MizBee \cite{meyer2009mizbee} are widely useful, especially in comparing two genomes, there are not many tools designed to explore more than two genomes at once. Because of projects such as the thousand genomes project \cite{1000genomes}, there is a demand for understanding the relationships among many genomes.
Adding one more axis of genome to make a 3D scatter plot, for example, may work in some (very limited) cases. We would see a line in diagonal in a comparison among human, chimpanzee and gorilla genomes\cite{synmap3durl}.
However, it is not visually intuitive to find out complicated relations in three species in general. 
In fact, Tory et al. \cite{tory2007spatialization} showed that a 3D landscape works not as well as a 2D map in specific tasks such as search and point estimation.
This casts doubt on whether three-dimensional techniques are useful for other tasks such as synteny visualiatization.
Moreover, a 3D scatterplot clearly does not generalize to more than three genomes.

%TODO citaion? what is paper first introduced scatterplot matrix?
Traditional scatterplot matrix displays can be readily adapted for comparing multiple genomes, with each subplot being a syntenic dotplot of two of the genomes.
However, scatterplot matrices do not scale up well beyond a moderate number of plots~\cite{}.
As the number of genomes increases, it becomes harder to tell the overall relation between the genomes.

In this poster, we describe an ongoing collaboration with domain scientists using multiple-genome comparisons in two use cases.
%
The first use case involves a comparison of about one hundred \emph{arabidopsis thaliana} genomes. In this case, the differences between the genomes are encoded by
a set of SNPs (single-nucleotide polymorphisms).
%
The second use case involves 17 genomes, each of a different species of the plasmodium genus. In this case, the genome differences are more complex, often architectural (that is, involving inversions and duplications of entire portions of the genome), and need to be encoded by the entire synteny map.

\begin{figure}[t]
 \centering
 \includegraphics[width=0.7\columnwidth]{synmap_n}
 \caption{A SynMapN plot of various species. Note the distant location of Escherichia coli (E.coli), the cluster of chimpanzee, gorilla and human, as well as the cluster of cat, dog and horse}
 \label{fig:synmap_n}
\end{figure}

\begin{figure}[h]
 \centering
 \includegraphics[width=0.8\columnwidth]{dotplot_arabidopsis_marked}
 \caption{A syntenic dotplot of Arabidopsis lyrata and Arabidopsis thaliana. The orange region 1 shows a duplication in Arabidopsis thaliana with respect to Arabidopsis lyrata. The blue region 2 shows an inversion. The image is regenerated by ks data downloaded from \cite{arabidopsisurl}.}
 \label{fig:dotplot_arabidopsis_marked}
\end{figure}

%\begin{figure}[h]
% \centering
% \includegraphics[width=\columnwidth]{lambda}
% \caption{lambda test}
% \label{fig:lambda}
%\end{figure}

 %TODO do this?
%\section{Related Works}
\section{Examples}
\autoref{fig:synmap_n} shows our first attempt of providing an overview of the genomes in concern. We define a similarity measure between two genomes, apply dimensionality reduction (Kernel PCA \cite{kernelpca}), and try to visualize the relationships among genomes of interest through a genomic map where distances between genomes encodes their affinity.

\begin{figure}[th]
 \centering
 \includegraphics[width=0.8\columnwidth]{chromosomes_matrix}
 \caption{Distance matrices of 97 Arabidopsis thaliana of different ecotypes. Eight subplots are 7 individual chromosomes and a sum of them. Clusters and outliers can be spotted in chromosome C (subplot \#6)}
 \label{fig:chromosomes_matrix}
\end{figure}

\begin{figure}[th]
 \centering
 \includegraphics[width=0.8\columnwidth]{chromosomes_scatter}
 \caption{MDS plots of 97 Arabidopsis thaliana of different ecotypes.}
 \label{fig:chromosomes_scatter}
\end{figure}

\autoref{fig:chromosomes_matrix} and \autoref{fig:chromosomes_scatter} show distance matrices and scatterplots (coordinates computed from MDS) of species from another dataset. The genomes are Arabidopsis of different ecotypes. When we break the entire genome into individual chromosomes and try to compute their pairwise edit distances, we see clusters in small chloroplast chromosomes (chrC, subplot \#6 in \autoref{fig:chromosomes_matrix} and \autoref{fig:chromosomes_scatter}). We can also see some common outliers among other plots.

%TODO to be trimed
\section{Method}
In the process of drawing a SynMap of two genomes, the program computes a measure called synonymous mutation rate ($ks$) describing a matching score for each pair of aligned genes between two genomes. This score ranges from $0$ to $+\infty$, or arbitrary large number in data, with $0$ means perfect alignment and infinity indicates no alignment or an error. We want to utilize these measures to build a notion of distance between two genomes. 

For example, when comparing two genomes of human and chimp, we are given IDs of gene pairs and their synonymous mutation rate, denoted by $ks$, which can be seen as a measure of distance (\autoref{tab:table_input_example})

\begin{table}[h!]
\centering
\caption{Data Input File}
\label{tab:table_input_example}
\begin{tabular}{cc|c}
\toprule
%\multicolumn{2}{c}{Gene labels} &\\
%\midrule
Human & Chimpanzee & $ks_i$\\
gene ID & gene ID & \\
\midrule
h1 & c1 & 0.0056\\
h2 & c2 & 72.6574\\
... & ... & ...	\\
\bottomrule
\end{tabular}
\end{table}

Now we want to find a function of many ks values that describes the distance between two genomes or equivalently, a notion of similarity or kernel between two entities.
\begin{align}
similarity(Human, Chimp) = g(ks_1,ks_2...)
\end{align}


Generally the gene IDs in such table may be duplicated, indicating a match of a single gene from one genome with multiple genes in the other genome (duplication). To formalize the computation, we put $ks$ values in matrix form

%ks_{2,1} & ks_{2,2} & ks_{2,3} & \dots & ks_{2, c}\\
%matrix representation
$$K = 
\begin{bmatrix}
ks_{1,1} & ks_{1,2} & \dots & ks_{1, c}\\
\vdots & \vdots & \ddots & \vdots \\
ks_{h,1} & ks_{h,2} & \dots & ks_{h, c}\\
\end{bmatrix}
$$

Where $ks_{1,1}$ stores the ks value between the first gene of human and the first gene of chimpanzee, and so on. The indices of rows and columns are in the order of gene locations in ordered chromosomes.
%, chromosomes are ordered in a natural order to genomicists. 
$h$ and $c$ are gene counts of human and chimpanzee respectively. One can think of $K$ as an image data of the SymMap plot (\autoref{fig:dotplot_arabidopsis_marked}).

Now we want a function of $K$ that outputs a scalar value to describe the similarity between two genomes.
\begin{align}
similarity(Human, Chimp) = f(K)
\end{align}

%TODO include this?
%Intuitively we want a measure similar to the dot product of vectors of unit length, which encodes cosine of the angle between the two vectors. Moreover, the dot products should better not generating any negative values since there is not a notion of 'negative of a human' among species.

To define a function so that comparison of a genome to itself gives a similarity measure close to $1$, we used $f$ to compute kernels between any two genomes, which are later used to plot \autoref{fig:synmap_n}
\begin{align}
f(K) = \frac{\sum_{i,j} e^{-\lambda K_{ij}}}{\sqrt{c \times h}}
\end{align}
Where $\lambda$ is a scalar constant related to sensitivity of the similarity measure. $c$ and $h$ are the gene counts defined before. 

Once we have the pairwise distances among multiple genomes (stored in matrix $M$), we have an implied space and then we project it onto a 2D screen using Kernel PCA.
Let $M$ be the matrix of similarities of multiple genomes where each entry is computed by function $f$ defined above. Note that this is a symmetric matrix with diagonal entries set to $1$, indicating complete identity in similarity measure of each genomes with respect to itself.
$$M = 
\begin{bmatrix}
f_{human, human} & f_{human, chimp} & f_{human,cat} & \dots \\
f_{chimp, human} & f_{chimp, chimp} & f_{chimp, cat} & \dots \\
\vdots & \vdots & \vdots & \ddots \\
\end{bmatrix}
$$
We then treat it as a kernel matrix and feed it into existing Kernel PCA algorithms and plot the first two principle components, which is shown in \autoref{fig:synmap_n}
\section{Other Examples}

\begin{figure}[h]
 \centering
 %\includegraphics[width=0.8\textwidth]{subset}
 \includegraphics[width=\columnwidth]{subset}
 \caption{A SynMapN plot with subsets of species concerned. The similarity measure of certain pairs are the same among all plots above so the distance encoded are consistent.}
 \label{fig:subset}
\end{figure}

In addition to the example plot in \autoref{fig:synmap_n}, we explore different subsets of the genomes among the genomes we concerned to show the capacities and constraints of SynMapN. 
%We show another distance measure used for a different dataset of more similar species represented in single-nucleotide polymorphism (SNP) data format.

%different lambda values and diferent subset of genomes
%closely related genomes, different approach

Plots in \autoref{fig:subset} shows subsets of the genomes we concerned with one genome excluded a time. The similarity of each pair of genomes are consistent among all the plots so the distances in the implied high dimensional space remains the same. But note that a 2D Kernel PCA plot reveals more of the true distances wist respect to an outlier through a certain plot, while potentially shrink the pairwise distances among other points. When an outlier is removed from the input of Kernel PCA in the next plot, another outlier shows up, indicating that previous plot did not show the true distance of the later outlier. For example, plot 2 shows affinity of horse, cat and dog, while with mouse excluded in plot 3, horse moves away from the cat-dog cluster. 

SynMapN gives an overview of genomes while ignoring the details of the comparison. The detailed patterns within two genomes comparison such as inversions and duplications are not shown in SynMapN. The other restriction lies on the limitation of dimensionality reduction method we use. Kernel PCA and Classical MDS\cite{mds} are essentially linear projections that preserves the pairwise distances globally, so the distant dots reflects the dissimilarities of the entities while the closed dots may only be an artifact of the projection, thus misleadingly reflects closeness of entities in similarity. So SynMapN at this stage is very useful in finding one specie that are most dissimilar from the others while the clustering it plots does not always reflect the truth.



%(more on SNPs dataset?)

  
\section{Future Works}
In a complete visualization system, we expect to enable users to navigate through various visualizations for different levels of details, from high level SynMapN to very detailed comparison of two specific genomes in SynMap. To further increase the number of levels, we can split genomes into individual chromosomes and compare them in SymMapN. We are also exploring the possibilities of capturing certain features in a SynMap plot, for example, the diagonal alignments or anti-diagonal alignments, using image processing. More improvements on interaction can be done through studying observation level interactions \cite{endert2011observation}, to enable users to specify desired clustering criteria through dragging points in the plot. 

%TODO do this?
%\section{Conlcusion}



%% if specified like this the section will be committed in review mode
\acknowledgments{
%The authors wish to thank A, B, C. This work was supported in part by
%a grant from XYZ.
}

%\bibliographystyle{abbrv}
\bibliographystyle{abbrv-doi}
%\bibliographystyle{abbrv-doi-narrow}
%\bibliographystyle{abbrv-doi-hyperref}
%\bibliographystyle{abbrv-doi-hyperref-narrow}

\bibliography{mybib}
\end{document}
